wasm2c/wasm-rt-impl.c

```diff
@@ -125,23 +164,37 @@ uint32_t wasm_rt_register_func_type(uint32_t param_count,
     func_type.results[i] = va_arg(args, wasm_rt_type_t);
   va_end(args);
 
-  for (i = 0; i < g_func_type_count; ++i)
-    if (func_types_are_equal(&g_func_types[i], &func_type))
-      return i + 1;
+  bool found = false;
```

Comment:
I guess you could have used `goto done;` to avoid this extra variable, and the extra indentation of the !found block below.. but this approach doesn't seem much worse.